### ISSUE TITLE
Fix the current course instance redirection when the newest instance is not open yet

### ISSUE THREAD

**Issue by Mankro** [apluslms/a-plus#919]
The current course instance redirection (URL contains course without any course instance identifier): https://plus.cs.aalto.fi/o1

https://github.com/apluslms/a-plus/blob/6a7704d22469cb0e2c0f1f52ba420cb3a4cf3c7c/course/views.py#L110

From @jsorva:

In a meeting some time ago, I promised to return to the issue of the 
“current instance link” that points to the most recent version of a 
course. So, here’s how I’d prefer like it to work:

1) While a course is running, the link points to the active instance. 
[For O1, this is in during the fall term.]

2) After a course is over [in December], the link continues to points to 
the most recent instance. This continues even after next year’s instance 
has been created in A+ but isn’t yet open to the public [typically early 
in the next calendar year]. And it even continues after the next 
instance is technically visible in some form but there’s only the front 
page and maybe some other materials, but the actual course contents 
aren’t visible yet [this is often the case during the summer].

3) Only when the actual course contents are visible [ideally some time 
in August, but might be September] does the link start pointing at the 
new instance.

To summarize, I’d like the link to always point to the newest instance 
whose actual contents are actually visible; the mere existence of the 
instance isn’t enough. (I’m not sure how close this wish is to the 
current behavior.)

Internal ticket: https://rt.cs.aalto.fi/Ticket/Display.html?id=20168

-----------------------------------

My first hunch is that this should be easy to accomplish by tweaking the latest course instance selection in the LastInstanceView.
